tools/custom_check.py

```diff
@@ -0,0 +1,279 @@
+        {'pattern': "== None",
+         'description': 'Use `is None` to check whether something is None'},
+        {'pattern': "type:[(]",
+         'description': 'Missing whitespace after ":" in type annotation'},
+        {'pattern': "# type [(]",
+         'description': 'Missing : after type in type annotation'},
+        {'pattern': "#type",
+         'description': 'Missing whitespace after "#" in type annotation'},
+        {'pattern': 'if[(]',
+         'description': 'Missing space between if and ('},
+        {'pattern': ", [)]",
+         'description': 'Unnecessary whitespace between "," and ")"'},
+        {'pattern': "%  [(]",
+         'description': 'Unnecessary whitespace between "%" and "("'},
+        # This next check could have false positives, but it seems pretty
+        # rare; if we find any, they can be added to the exclude list for
+        # this rule.
+        {'pattern': ' % [a-zA-Z0-9_.]*\)?$',
+         'description': 'Used % comprehension without a tuple'},
+        {'pattern': '.*%s.* % \([a-zA-Z0-9_.]*\)$',
+         'description': 'Used % comprehension without a tuple'},
+        {'pattern': 'json_success\({}\)',
+         'description': 'Use json_success() to return nothing'},
+        # To avoid json_error(_variable) and json_error(_(variable))
+        {'pattern': '\Wjson_error\(_\(?\w+\)',
+         'exclude': set(['zerver/tests']),
+         'description': 'Argument to json_error should be a literal string enclosed by _()'},
+        {'pattern': '\Wjson_error\([\'"].+[),]$',
+         'description': 'Argument to json_error should a literal string enclosed by _()'},
+        # To avoid JsonableError(_variable) and JsonableError(_(variable))
+        {'pattern': '\WJsonableError\(_\(?\w.+\)',
+         'exclude': set(['zerver/tests']),
+         'description': 'Argument to JsonableError should be a literal string enclosed by _()'},
+        {'pattern': '\WJsonableError\(["\'].+\)',
+         'description': 'Argument to JsonableError should be a literal string enclosed by _()'},
+        {'pattern': '([a-zA-Z0-9_]+)=REQ\([\'"]\\1[\'"]',
+         'description': 'REQ\'s first argument already defaults to parameter name'},
+        {'pattern': 'self\.client\.(get|post|patch|put|delete)',
+         'description': \
+         '''Do not call self.client directly for put/patch/post/get.
+    See WRAPPER_COMMENT in test_helpers.py for details.
+    '''},
+        # Directly fetching Message objects in e.g. views code is often a security bug.
+        {'pattern': '[^r][M]essage.objects.get',
+         'description': 'Please use access_message() to fetch Message objects',
+         },
+        {'pattern': 'render_to_response\(',
+         'description': "Use render() instead of render_to_response().",
+         },
+        # This rule might give false positives in virtualenv setup files which should be excluded,
+        # and comments which should be rewritten to avoid use of "python2", "python3", etc.
```

Comment:
I think the ~10 rules from here up (first one is `json_success`) are all unnecessary in the bots repo and we should perhaps remove.